File: ARC/ARC029/a.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    int t[N];
    for (int i=0; i<N; ++i) cin >> t[i];

    int ans = 100000;
    for (int i=0; i<(1<<N); ++i) {
        int t1=0, t2=0;
        for (int j=0; j<N; ++j) {
            if (i >> j & 1) t1 += t[j];
            else t2 += t[j];
        }
        ans = min(ans, max(t1, t2));
    }
    cout << ans << endl;

    return 0;
}

File: ARC/ARC029/a.py
N = int(input())
t = [int(input()) for _ in range(N)]

ans = 10 ** 9
for i in range(2 ** N):
    niku1 = 0
    niku2 = 0
    for j in range(N):
        if (i >> j) & 1:
            niku1 += t[j]
        else:
            niku2 += t[j]
    if ans > max([niku1, niku2]):
        ans = max([niku1, niku2])
print(ans)